scripts/test-unit.js

```diff
@@ -0,0 +1,35 @@
+// This file exists because mocha doesn't return non-zero value when there is a failed test.
+// When https://github.com/mochajs/mocha/issues/3893 is fixed, it will be removed.
+
+const { spawn } = require('child_process')
+
+// Test result on console.
+let log = ''
+const proc = spawn(`node`, ['./node_modules/.bin/mocha', 'src/**/*.spec.*'], {
+})
+
+proc.stdout.on('data', (data) => {
+  log += data
+})
+
+proc.stdout.on('end', async () => {
+  await sleep(500)
+
+  if (log.match(/[0-9]+ passing.*\n\s*[0-9]+ failing/g) !== null) {
+    process.exit(1)
+  }
+
+  process.exit(0)
+})
+
+function sleep (ms) {
```

Comment:
Pull in Bluebird for the Promise (instead of the native one) and use [Promise.delay](http://bluebirdjs.com/docs/api/promise.delay.html) instead.